agent/resources/resources_linux.go

```diff
@@ -0,0 +1,119 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package resources
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+// resource to implement the Resources interface
+type resources struct {
+	control cgroup.Control
+}
+
+// New is used to return an object of the resources
+func New() Resources {
+	return newResources(cgroup.New())
+}
+
+func newResources(control cgroup.Control) Resources {
+	return &resources{
+		control: control,
+	}
+}
+
+// Init is used to initialize the resources
+func (r *resources) Init() error {
+	return r.cgroupInit()
+}
+
+// Setup sets up the resources
+func (r *resources) Setup(task *api.Task) error {
+	return r.setupCgroup(task)
+}
+
+// Cleanup removes the resources
+func (r *resources) Cleanup(task *api.Task) error {
+	return r.cleanupCgroup(task)
+}
+
+// cgroupInit is used to create the root '/ecs/ cgroup
+func (r *resources) cgroupInit() error {
```

Comment:
I'm not sure cgroup specific functions belong here, but I'm not sure where else to put them.